src/components/TextField/TextField.st.css

```diff
@@ -0,0 +1,185 @@
+    MainTextFont: --overridable;
+}
+
+:vars {
+    mainBorderColor: color(fallback(value(MainBorderColor), value(defaultMainBorderColor)));
+    mainBackgroundColor: color(fallback(value(MainBackgroundColor), value(defaultMainBackgroundColor)));
+    mainTextColor: color(fallback(value(MainTextColor), value(defaultMainTextColor)));
+    mainTextFont: fallback(font(value(MainTextFont)), font(value(defaultMainTextFont)));
+}
+
+:vars {
+    /** default state */
+    textColor: value(mainTextColor);
+    colorPlaceholder: applyOpacity(value(mainTextColor), 0.6);
+    backgroundColor: value(mainBackgroundColor);
+    borderColor: value(mainBorderColor);
+    border: 1px solid applyOpacity(value(borderColor), 0.6);
+    successBorderColor: applyOpacity(value(successColor), 0.6);
+    errorBorderColor: applyOpacity(value(errorColor), 0.6);
+}
+
+:vars {
+    /*Hover*/
+    colorHover: value(mainTextColor);
+    backgroundColorHover: value(mainBackgroundColor);
+    borderColorHover: value(borderColor);
+    successBorderColorHover: value(successColor);
+    errorBorderColorHover: value(errorColor);
+}
+
+:vars {
+    /*Focus*/
+    colorFocus: value(mainTextColor);
+    backgroundColorFocus: value(mainBackgroundColor);
+    borderColorFocus: value(borderColor);
+    successBorderColorFocus: value(successColor);
+    errorBorderColorFocus: value(errorColor);
+}
+
+:vars {
+    /*Disabled*/
+    colorDisabled: applyOpacity(value(mainTextColor), 0.6);
+    backgroundColorDisabled: value(mainBackgroundColor);
+    borderColorDisabled: applyOpacity(value(mainBorderColor), 0.6);
+}
+
+.root {
+    -st-extends: Input;
+    -st-states: error, success, theme(enum(line, box));
+    box-sizing: border-box;
+    border-color: value(borderColor);
+    border: value(border);
+    border-radius: 0;
+    transition: border-color .1s ease-in;
+    background-color: value(mainBackgroundColor);
+    /* todo - Define fonts somewhere*/
+    font: value(mainTextFont);
+    display: flex;
+    align-items: center;
+    padding: 8px 12px;
+}
+
+.root:hover {
+    border-color: value(borderColorHover);
+    background-color: value(backgroundColorHover);
+}
+
+.root:focus {
+    border-color: value(borderColorFocus);
+    background-color: value(backgroundColorFocus);
+}
+
+.root:disabled {
+    border-color: value(borderColorDisabled);
+    background-color: value(backgroundColorDisabled);
+}
+
+.root:success {
+    border-color: value(successBorderColor);
+}
+
+.root:hover:success {
+    border-color: value(successBorderColorHover);
+}
+
+.root:focus:success {
+    border-color: value(successBorderColorFocus);
+}
+
+.root:error {
+    border-color: value(errorBorderColor);
+}
+
+.root:hover:error {
+    border-color: value(errorBorderColorHover);
+}
+
+.root:focus:error {
+    border-color: value(errorBorderColorFocus);
+}
+
+.root:theme(line) {
+    border-top: none;
+    border-left: none;
+    border-right: none;
+    padding: 8px 0;
+    background-color: transparent;
```

Comment:
Designer QA note:
the `line` theme should have a background-color that can be overridable, like this:
```background-color: color(fallback(MainBackgroundColor, transparent))``` 
Note the `MainBackgroundColor` with the big `M`

Reply:
Done.